### PR TITLE
adding apps methods and tests

### PR DESCRIPTION
**Pull request by pariosur**
Addition to SDK
- add_app()
- get_app()
- get_apps()

### CODE REVIEW

File: src/metal_sdk/metal.py [hunk 1]
```diff
@@ -413,3 +414,25 @@ def get_queries(self, index_id: str) -> dict:
...
+        if not app_id:
+            raise TypeError("app_id required")
+        if len(app_id) != 24:
+            raise ValueError("app_id must have a length of 24 characters")
+
+        url = f"/v1/apps/{app_id}"
+        res = self.fetch("get", url, None)
+        return res
+
+    def get_all_apps(self) -> dict:
```
[the remark this answers is not in the document]
Comment (pariosur) [reply src/metal_sdk/metal.py:434]: I started using it with datasources and data_entities. Should we instead use `get_apps()` and `get_app()`?

**Comment by pariosur** [comment]
@softboyjimbo changes are done 💯

File: examples/example.py [hunk 2]
```diff
@@ -15,7 +15,9 @@ def main():
     # metal.index({ "id": "667", "text": "Dave Mustain" })
     # metal.index({ "id": "666", "text": "Ozzy" })
     # metal.index({ "id": "668", "text": "Bruce Dickenson" })
-    metal.search({"text": "908", "limit": 1})
+    # metal.search({"text": "908", "limit": 1})
+    results = metal.get_all_apps()
```
Comment (softboyjimbo) [inline examples/example.py:19]: update to `get_apps()`

File: tests/test_metal_async.py [hunk 3]
```diff
@@ -510,3 +510,47 @@ async def test_metal_get_queries(self):
...
+        return_value = mock.MagicMock(json=lambda: mock_response_data)
+        metal.request = mock.AsyncMock(return_value=return_value)
+
+        await metal.get_app(mock_app_id)
+
+        self.assertEqual(metal.request.call_count, 1)
+        self.assertEqual(metal.request.call_args[0][0], "get")
+        self.assertEqual(metal.request.call_args[0][1], f"/v1/apps/{mock_app_id}")
+
+    async def test_metal_get_all_apps(self):
```
Comment (softboyjimbo) [inline tests/test_metal_async.py:549]: update naming to `get_apps`

File: tests/test_metal.py [hunk 4]
```diff
@@ -545,3 +545,47 @@ def test_metal_get_queries(self):
...
+        metal.request = mock.MagicMock(return_value=mock.Mock(status_code=201))
+        metal.add_app(payload)
+
+        self.assertEqual(metal.request.call_count, 1)
+
+        self.assertEqual(metal.request.call_args[0][0], "post")
+        self.assertEqual(metal.request.call_args[0][1], "v1/apps")
+        self.assertEqual(metal.request.call_args[1]["json"]["name"], mock_app_name)
+
+    def test_metal_get_all_apps(self):
```
Comment (softboyjimbo) [inline tests/test_metal.py:565]: update naming to `get_apps`

Comment (pariosur) [reply tests/test_metal.py]: sorry I missed this! now it should be fully removed

**Review by softboyjimbo — APPROVED** [review]

**Pull request merged**